Addition of MsBackendMzr saving and loading. Discussion needed

philouail opened the pull request:
Alright so here is the code that allows to save and load an `MsBakendMzR` object. This refer to #2 

Few topics of discussion: 

- Organization of the code and the documentation: I think this works okay and should be fine for the future addition.
- Dependence on Spectra import: @jorainer mentioned not to do that because i need many function from that package, maybe I misunderstood something ? 
- I think this short code is also a great example on where we could utilize the `alabaster` package, instead of using `writeLines()` and `read.table()` to import and export the `spectraData`. @lgatto  is that what you had in mind ? 
- I had to do a fix line 60 of `MsBackendMzR.R` file because the initial code to `sub()` the `dataStorage()` of the object did not work in my windows system. I don't know if it's the best way to do it. 

Once this is pushed I will continue to move and change the code for saving/loading a `Spectra` object,...

jorainer commented:
Regarding the *alabastter* issues/errors above: you should use `readObject()` instead of `loadObject()` :)

philouail commented:
> Regarding the _alabastter_ issues/errors above: you should use `readObject()` instead of `loadObject()` :)

Well i should have figured that out, thaanks :)

jorainer reviewed (changes requested):
Thanks Phili! Great and clean code. I just have some minor suggestions - and some more to discuss. In particular:

- I would not mix `PlainTextParam` export with e.g. `AlabasterParam`, but offer/implement the two approaches separately.
- I would suggest to keep the `PlainTextParam` as you have it here and in addition implement *alabaster*'s `saveObject,MsBackendMzR` and `readObject,MsBackendMzR`. These could then either directly used, or through a `saveMsObject,MsBackendMzR,AlabasterParam` and `readMsObject,MsBackendMzR,AlabasterParam`.
- WHY? a) to directly compare how they look and feel. b) for integration with external software, IMHO, *alabaster*'s HDF5-based file formats might not be easy to use. I would hence still like to support a pure text file format (note that this is then mostly for the *higher level* result objects such as `XcmsExperiment` and `SummarizedExperiment`).

philouail commented:
Hi @jorainer, i've updated to fix the dependency. Can you check again and merge if it's all good ? also I cannot see the comments you made during our meeting but i think i fixed everything, do tell me if i am missing something.

jorainer commented on R/MsBackendMzR.R:
```diff
@@ -0,0 +1,66 @@
+setMethod("readMsObject", signature(object = "MsBackendMzR",
+                                   param = "PlainTextParam"),
+          function(object, param, spectraPath = character()) {
+              b <- Spectra::MsBackendMzR()
```
`object` is already an instance of a `MsBackendMzR` class, so, instead of `b` you could just use `object` throughout the code of this function.

jorainer commented on R/PlainTextParam.R:
```diff
@@ -0,0 +1,66 @@
+#' @description
+#'
+#' The `saveMsObject()` and `readMsObject()` methods with the `PlainTextParam`
+#' option enable users to save/load different type of Ms object as a
```
Please change *Ms* with *MS*

jorainer commented on R/PlainTextParam.R:
```diff
@@ -0,0 +1,66 @@
+#' function. Any previous exports eventually present in that folder will be
+#' overwritten.
+#'
+#' The Ms object currently supported for import and export with this parameter
```
Also here, please change *Ms* with *MS*.

jorainer reviewed (approved):
Great! Thanks!
